libsolidity/formal/SMTEncoder.cpp

```diff
@@ -861,6 +857,9 @@ void SMTEncoder::endVisit(Identifier const& _identifier)
 		defineExpr(_identifier, m_context.state().thisAddress());
 		m_uninterpretedTerms.insert(&_identifier);
 	}
+	// Ignore struct type identifiers in struct constructor calls
+	else if (auto typetype = dynamic_cast<TypeType const*>(_identifier.annotation().type); typetype && typetype->actualType()->category() == Type::Category::Struct)
```

Comment:
```suggestion
	else if (auto typetype = dynamic_cast<TypeType const*>(_identifier.annotation().type);
	typetype && typetype->actualType()->category() == Type::Category::Struct)
```

Reply:
So the next PR just needs to change this if to the dynamic cast being not null, right?

Reply:
I believe so. When I tried it it removed a lot of warnings about SMTChecker not supporting a type, and I believe it did not have any negative effect on other tested features.
I was just not sure if all the situations containing a type identifier are now supported, or if there are some which are not, and we lose this information.

Reply:
I think they should all be supported.
So please just apply the style changes to this PR, and we can change the if in the next PR.